Add a Github action file to test runnable_cxx

Geod24 opened the pull request:
```
This allows us to test `runnable_cxx` with multiple versions of g++ and clang on linux,
and multiple versions of clang on OSX.
In the future, we can extend it to work on MSVC as well,
but in its current form this can unblock PR 10927.
```

This should allow #10927 to move forward with proper tests. The `runnable_cxx` currently still runs on the auto-tester, but if this proves to work well, we could disable it on Mac / Linux.

CC @MoonlightSentinel 

~EDIT: Ah, this is not checking out the proper branch for druntime / phobos when testing stable. Will fix.~

Geod24 committed:
Add a Github action file to test runnable_cxx

This allows us to test `runnable_cxx` with multiple versions of g++ and clang on linux,
and multiple versions of clang on OSX.
In the future, we can extend it to work on MSVC as well,
but in its current form this can unblock PR 10927.

Geod24 commented:
CC @wilzbach @WalterBright : Could you check if actions are enabled for this repository ?
You can check here: https://github.com/dlang/dmd/settings/actions
Here's the result of the run from my fork: https://github.com/Geod24/dmd/actions/runs/62718036

MoonlightSentinel commented on .github/workflows/runnable_cxx.yml:
```diff
@@ -0,0 +1,262 @@
+      # https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners#supported-runners-and-hardware-resources
+      run: |
+        # All hosts are 64 bits but let's be explicit
+        ./dmd/src/build.d MODEL=64
```
Yould could be explicit for `build.d` and `run.d` as well (allthough they should probably detect the correct value anyway)

```suggestion
        ./dmd/src/build.d -j2 MODEL=64
```

MoonlightSentinel commented on .github/workflows/runnable_cxx.yml:
```diff
@@ -0,0 +1,262 @@
+    ########################################
+    - name: Run C++ test suite
+      run: |
+        ./dmd/test/run.d runnable_cxx MODEL=64
```
The environment should be available in the logs:
```suggestion
        ./dmd/test/run.d --environment runnable_cxx MODEL=64
```

MoonlightSentinel replied:
This CI might even be a candidate for `TRY_DISABLED` (depending on the average runtime)

Geod24 commented:
Addressed review. Ping @WalterBright for the admin thing. This will allow us to have much better cross-platform testing, which we really need. I keep finding bugs in `extern(C++)`, just today we found 2 in the span of an hour...

PetarKirov commented:
> CC @wilzbach @WalterBright : Could you check if actions are enabled for this repository ?

Also CC @CyberShadow @atilaneves

wilzbach commented:
It's already on allowed:

![image](https://user-images.githubusercontent.com/4370550/77665554-1c543880-6f80-11ea-819a-1bc85c694978.png)

Geod24 commented:
Fixed the base branch checkout. Shall we move forward then ?
